sdks/python/apache_beam/io/filebasedsource.py

```diff
@@ -207,22 +212,36 @@ def split(self, desired_bundle_size, start_offset=None, stop_offset=None):
     if stop_offset is None:
       stop_offset = self._stop_offset
 
-    bundle_size = max(desired_bundle_size, self._min_bundle_size)
-
-    bundle_start = start_offset
-    while bundle_start < stop_offset:
-      bundle_stop = min(bundle_start + bundle_size, stop_offset)
+    if self._file_based_source.splittable:
+      bundle_size = max(desired_bundle_size, self._min_bundle_size)
+
+      bundle_start = start_offset
+      while bundle_start < stop_offset:
+        bundle_stop = min(bundle_start + bundle_size, stop_offset)
```

Comment:
Range [start_offset, stop_offset) is the range for all bundles. So stop_offset should be the stop position of the last bundle.